### ISSUE TITLE
[Question] How did you compute the GFLOPs in your paper?

### ISSUE THREAD

**Issue by ConnollyLeon** [Dao-AILab/flash-attention#741]
![image](https://github.com/Dao-AILab/flash-attention/assets/31810583/856cc5c5-a2e4-4f96-b52a-bf7199900300)

Hi, I wonder how you compute the FLOPs based on this setting: (seq. length 1024, head dim. 64, 16 heads, batch size 64).

I try to calculate it using formula like 4* seq_length^2 * head_num * heads * batch_size, but it easily surpass 66.6 GFLOPs.

**Comment by tridao** [comment]
Try decreasing the batch size?

**Comment by ConnollyLeon** [comment]
> Try decreasing the batch size?

Hmmm, That would be 4 GFLOPs I guess. Did I miss something? Even by adding the FLOPs of backward pass (i.e.,  additional 8 GFLOPs), there is still a huge difference between 12 and 66.6.

**Comment by tridao** [comment]
Yeah the FLOPS numbers might have been off by some factor. The ratio between the two columns should still be around the same.

**Comment by tridao** [comment]
I think forward FLOPs (just counting matmuls) is 4* seq_length^2 * head_num * heads * batch_size = 275GFLOPs. Backward is 2x that for standard attention and 2.5x that for FlashAttention. 
So the left column FLOPs should be 825 GFLOPs and the right column should be 963 GFLOPs.

**Comment by ConnollyLeon** [comment]
Thanks for your reply. But the ratio is subtly different from the paper.

75.2/66.6 ≈ 1.129
963/825 ≈ 1.167

What did we miss?

**Comment by tridao** [comment]
I think I was adding some FLOPs from the softmax and masking which are much smaller compared to the matmul FLOPs. For simplicity I think it's easier to just calculate the matmul FLOPs.
